.github/workflows/snapshot-test.yml

```diff
@@ -0,0 +1,99 @@
+name: QA - Snapshot Test
+
+on:
+  workflow_dispatch:
+
+jobs:
+  execution-test-suite:
+    runs-on: [ self-hosted, Erigon3 ] # must run on E2 not to steal E3 runner, which is used for PR approval
+    timeout-minutes: 7200  # 5 days
+    env:
+      ERIGON_QA_PATH: /opt/erigon-qa
+      ERIGON_DATA_DIR: /opt/erigon-versions/reference-version/datadir
+      ERIGON_SNAPSHOT_TEST_DIR: /opt/erigon-snapshot
+      CHAIN: mainnet
+      TRACKING_TIME_SECONDS: 431400  # 5 days minus 10 minutes
+      TOTAL_TIME_SECONDS: 431400
+
+    steps:
+      - name: Checkout Silkworm Repository
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: "0"
+
+      - name: Clean Build Directory
+        run: rm -rf ${{runner.workspace}}/silkworm/build
+
+      - name: Create Build Environment
+        run: cmake -E make_directory ${{runner.workspace}}/silkworm/build
+
+      - name: Configure CMake
+        working-directory: ${{runner.workspace}}/silkworm/build
+        run: |
+          cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=Release
+
+      - name: Build Silkworm
+        working-directory: ${{runner.workspace}}/silkworm/build
+        run: cmake --build . --config Release --target silkworm -j 8
+
+      - name: Pause the Erigon instance dedicated to db maintenance
+        run: |
+          python3 $ERIGON_QA_PATH/test_system/db-producer/pause_production.py || true
+
+      - name: Sync the snapshot directory
+        run: |
+            rsync -a --delete $ERIGON_DATA_DIR/snapshots/ $ERIGON_SNAPSHOT_TEST_DIR/
+
+      - name: Run Execution Test
+        id: test_step
+        working-directory: ${{runner.workspace}}/silkworm/build/cmd
+        run: |
+          set +e # Disable exit on error
+          
+          # Run the test script
+          some_test_cmd
```

Comment:
We can start running some subcommands of `snapshots` utility as tests:
```
./snapshots --snapshot_dir $ERIGON_SNAPSHOT_TEST_DIR count_headers
./snapshots --snapshot_dir $ERIGON_SNAPSHOT_TEST_DIR count_bodies
```

Reply:
ok